phoenix-core/src/it/java/org/apache/phoenix/iterate/PhoenixQueryTimeoutIT.java

```diff
@@ -202,6 +204,57 @@ public void testScanningResultIteratorQueryTimeoutForPagingWithVeryLowTimeout()
         }
     }
 
+    @Test
+    public void testScanningResultIteratorShouldNotQueryTimeoutForPagingAfterReceivingAValidRow() throws Exception {
+        //Arrange
+        PreparedStatement ps = loadDataAndPreparePagedQuery(10,1);
+
+        ManualEnvironmentEdge injectEdge;
+        injectEdge = new ManualEnvironmentEdge();
+        injectEdge.setValue(EnvironmentEdgeManager.currentTimeMillis());
+        EnvironmentEdgeManager.injectEdge(injectEdge);
+
+        //Act + Assert
+        try {
+            //Do not let BaseResultIterators throw Timeout Exception Let ScanningResultIterator handle it.
+            BaseResultIterators.setForTestingSetTimeoutToMaxToLetQueryPassHere(true);
+            ResultSet rs = ps.executeQuery();
+            while(rs.next()) {
+                injectEdge.incrementValue(10);
+            }
+        } catch (SQLException e) {
+            fail("Query should have run smoothly");
+        } finally {
+            BaseResultIterators.setForTestingSetTimeoutToMaxToLetQueryPassHere(false);
+        }
+
```

Comment:
Please add more readable comments here. For example, this query should time out as the query timeout value is set to 0.